### DISCUSSION TITLE
How to add release notes etc. to commit bodies?

### DISCUSSION THREAD

**Discussion by mmkal** [renovatebot/renovate#30727] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi! I've got a config that pushes commits directly to a `deps` branch, which I later merge in via a separate process to `main` (pull requests are opened only when an update fails CI).

Something I miss from this setup, though, is the extra information that usually comes on a renovate pull request with links to release notes etc. I am trying to tell renovate to include this using `renovate.json` but what I'm trying doesn't seem to be working - all the template variables I use on `commitBody` have no effect. My current config is below ([linked here](https://github.com/mmkal/trpc-cli/blob/main/renovate.json)), but I've also tried `"commitBody": "{{releases}}"` to no avail:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "baseBranches": ["deps"],
  "automerge": true,
  "automergeType": "branch",
  "extends": [
    "config:recommended"
  ],
  "commitBody": "{{{header}}}{{{table}}}{{{notes}}}{{{changelogs}}}{{{configDescription}}}{{{footer}}}"
}
```

The commit remains a one liner like `chore(deps): update dependency tsx to v4.17.0` ([link](https://github.com/mmkal/trpc-cli/commit/de109c02fcdf8e9569dd198880cf45d6fcbdbda4))

What's the best way to make the commit body look like what the PR body usually looks like, or at least get a shortened version of the changelogs?

### Logs (if relevant)

Example job log: https://developer.mend.io/github/mmkal/trpc-cli/-/job/9f0c2124-b517-4440-8c82-2789a55e72c0

**Comment by RahulGautamSingh** [comment, marked as answer]
1. Set [fetchChangelogs](https://docs.renovatebot.com/configuration-options/#fetchchangelogs) to `branch`.
2. Read the availble template fields [here](https://docs.renovatebot.com/templates/#other-available-fields). (you will need to use `logJSON`)
3. Refer this [reproduction](https://github.com/mikebryant/renovate-test-changelogjson).

**Reply by mmkal** [reply]
Thank you, this worked! https://github.com/mmkal/trpc-cli/pull/31/commits/50ee7a35e02b4e62df79a4d40467fb1ed31064ed

**Reply by dietergeerts** [reply]
@RahulGautamSingh , is it possible to do this only for certain dependencies and not all? We have some repos that sometimes wrap things, and only the changelog of the wrapped things should go further down the line.
